infrastructure/service-bus.tf

```diff
@@ -29,21 +19,29 @@ resource "azurerm_servicebus_subscription_rule" "topic_filter_rule_sscs" {
   }
 }
 
-data "azurerm_key_vault" "hmc" {
+data "azurerm_key_vault" "hmc-key-vault" {
   name                = "hmc-${var.env}"
-  resource_group_name = local.resource_group_name
+  resource_group_name = "hmc-shared-${var.env}"
 }
 
-data "azurerm_key_vault_secret" "hmc" {
-  for_each     = toset(local.key_vault_secrets)
-  key_vault_id = data.azurerm_key_vault.hmc.id
-  name         = each.key
+data "azurerm_key_vault_secret" "hmc-servicebus-connection-string" {
+  key_vault_id = data.azurerm_key_vault.hmc-key-vault.id
+  name         = "hmc-servicebus-connection-string"
 }
 
-resource "azurerm_key_vault_secret" "shared_access_key" {
-  for_each     = data.azurerm_key_vault_secret.hmc
-  name         = each.value.name
-  value        = each.value.value
+resource "azurerm_key_vault_secret" "hmc-servicebus-connection-string" {
+  name         = "hmc-servicebus-connection-string"
+  value        = data.azurerm_key_vault_secret.hmc-servicebus-connection-string.value
   key_vault_id = data.azurerm_key_vault.sscs_key_vault.id
 }
 
+data "azurerm_key_vault_secret" "hmc-servicebus-shared-access-key" {
+  key_vault_id = data.azurerm_key_vault.hmc-key-vault.id
+  name         = "hmc-servicebus-shared-access-key"
+}
+
+resource "azurerm_key_vault_secret" "sscs-hmc-servicebus-hared-access-key" {
```

Comment:
There is this typo here "-hared-access-key"

Reply:
This is what the resource was called before, so I am keeping it the same.

Reply:
That's what I suspected